site/gatsby-site/src/components/forms/PreviewImageInputGroup.js

```diff
@@ -20,75 +23,111 @@ export default function PreviewImageInputGroup({
   alt = '',
   icon,
 }) {
-  const [cloudinaryID, setCloudinaryID] = useState(cloudinary_id);
+  const [cloudinaryID, setCloudinaryID] = useState(IMG_FALLBACK);
+
+  // Track whether the image is waiting to update so we can show a spinner.
+  const [updatingImage, setUpdatingImage] = useState(false);
+
+  const [imageReferenceError, setImageReferenceError] = useState(false);
+
+  const timeoutID = useRef(null);
 
-  // Debounced function needs to be in ref
-  // so it can maintain its internal state
-  const debouncedUpdateCloudinaryId = useRef(debounce((c) => setCloudinaryID(c), 2000)).current;
+  const imageUrl = useRef(values.media_url);
+
+  const isVideo = (url) => {
```

Comment:
This function is duplicated on `/src/components/cite/VideoPlayer.js`. Please refactor this to use just one function.